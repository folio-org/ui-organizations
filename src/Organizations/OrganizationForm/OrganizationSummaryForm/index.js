export { default as OrganizationSummaryForm } from './OrganizationSummaryForm';
