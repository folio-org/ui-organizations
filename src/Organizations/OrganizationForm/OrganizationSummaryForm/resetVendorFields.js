export default [
  'paymentMethod',
  'vendorCurrencies',
  'claimingInterval',
  'discountPercent',
  'expectedActivationInterval',
  'expectedInvoiceInterval',
  'expectedReceiptInterval',
  'renewalActivationInterval',
  'subscriptionInterval',
  'taxId',
  'taxPercentage',
  'liableForVat',
  'agreements',
  'edi.ediJob.schedulingNotes',
  'edi.ediJob.notifyErrorOnly',
  'edi.ediJob.notifyInvoiceOnly',
  'edi.ediJob.notifyAllEdi',
  'edi.ediJob.sendToEmails',
  'edi.ediJob.isSunday',
  'edi.ediJob.isSaturday',
  'edi.ediJob.isFriday',
  'edi.ediJob.isThursday',
  'edi.ediJob.isWednesday',
  'edi.ediJob.isTuesday',
  'edi.ediJob.isMonday',
  'edi.ediJob.time',
  'edi.ediJob.date',
  'edi.ediJob.scheduleEdi',
  'edi.ediFtp.notes',
  'edi.ediFtp.invoiceDirectory',
  'edi.ediFtp.orderDirectory',
  'edi.ediFtp.ftpPort',
  'edi.ediFtp.ftpConnMode',
  'edi.ediFtp.ftpMode',
  'edi.ediFtp.password',
  'edi.ediFtp.username',
  'edi.ediFtp.ftpFormat',
  'edi.ediFtp.serverAddress',
  'edi.notes',
  'edi.supportInvoice',
  'edi.supportOrder',
  'edi.sendAcctNum',
  'edi.ediNamingConvention',
  'edi.prorateFees',
  'edi.prorateTax',
  'edi.libEdiType',
  'edi.libEdiCode',
  'edi.vendorEdiType',
  'edi.vendorEdiCode',
  'edi.vendorEdiType',
  'edi.vendorEdiCode',
  'accounts',
];
