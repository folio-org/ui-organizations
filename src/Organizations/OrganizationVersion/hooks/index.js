export { useOrganizationVersions } from './useOrganizationVersions';
export { useSelectedOrganizationVersion } from './useSelectedOrganizationVersion';
