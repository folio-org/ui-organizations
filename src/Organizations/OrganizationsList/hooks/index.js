export * from './useBuildQuery';
export * from './useOrganizations';
