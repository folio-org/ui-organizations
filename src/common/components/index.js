export { default as AddressInfo } from './AddressInfo';
export { default as ButtonIsPrimary } from './ButtonIsPrimary';
export { default as EntitiesWithCollapsing } from './EntitiesWithCollapsing';
export { default as FieldIsPrimary } from './FieldIsPrimary';
export { default as FieldLanguage } from './FieldLanguage';
