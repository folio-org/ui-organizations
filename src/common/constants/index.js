export * from './api';
export * from './categories';
export * from './events';
export * from './interfaces';
export * from './organization';
export * from './organizationTypes';
export * from './routes';
export * from './sections';
export * from './vendorCategories';
