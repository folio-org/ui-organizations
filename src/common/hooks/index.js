export * from './useIntegrationConfig';
export * from './useIntegrationConfigMutation';
export * from './useTranslatedCategories';
export * from './useAcqMethods';
