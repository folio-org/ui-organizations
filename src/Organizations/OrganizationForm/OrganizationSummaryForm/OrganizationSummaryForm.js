import React, { useCallback, useState } from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import { Field, useForm } from 'react-final-form';
import { FieldArray } from 'react-final-form-arrays';

import {
  Checkbox,
  Col,
  ConfirmationModal,
  RepeatableField,
  Row,
  Select,
  TextArea,
  TextField,
} from '@folio/stripes/components';
import {
  AcqUnitsField,
  validateRequired,
} from '@folio/stripes-acq-components';

import { ORGANIZATION_STATUS } from '../../../common/constants';
import { FieldLanguage } from '../../../common/components';
import {
  CREATE_UNITS_PERM,
  MANAGE_UNITS_PERM,
} from '../../constants';
import resetVendorFields from './resetVendorFields';
import FieldCode from './FieldCode';

function OrganizationSummaryForm({ initialValues }) {
  const [isVendorUncheckConfirm, setVendorUncheckConfirm] = useState(false);
  const { change } = useForm();

  const renderAlias = useCallback((elem) => {
    return (
      <Row>
        <Col
          xs
          data-test-alias
        >
          <Field
            component={TextField}
            fullWidth
            label={<FormattedMessage id="ui-organizations.summary.alias" />}
            name={`${elem}.value`}
            required
            validate={validateRequired}
            validateFields={[]}
          />
        </Col>
        <Col xs>
          <Field
            component={TextField}
            fullWidth
            label={<FormattedMessage id="ui-organizations.summary.description" />}
            name={`${elem}.description`}
            validateFields={[]}
          />
        </Col>
      </Row>
    );
  }, []);

  const hideVendorUncheckConfirm = useCallback(() => {
    setVendorUncheckConfirm(false);
    change('isVendor', true);
  }, [change]);

  const onResetVendorFields = useCallback(() => {
    resetVendorFields.forEach(field => change(field, null));
  }, [change]);

  const handleVendorUncheck = useCallback(() => {
    setVendorUncheckConfirm(false);
    onResetVendorFields();
  }, [onResetVendorFields]);

  const onChangeIsVendor = useCallback(({ target: { checked } }) => {
    change('isVendor', checked);

    if (initialValues.id && !checked) setVendorUncheckConfirm(true);
  }, [initialValues.id, change]);

  const isEditMode = Boolean(initialValues.id);

  return (
    <Row>
      <Col
        xs={6}
        md={3}
      >
        <Field
          component={TextField}
          fullWidth
          id="name"
          label={<FormattedMessage id="ui-organizations.summary.name" />}
          name="name"
          required
          validate={validateRequired}
          validateFields={[]}
        />
      </Col>
      <Col
        xs={6}
        md={3}
      >
        <FieldCode orgId={initialValues.id} />
      </Col>
      <Col
        xs={6}
        md={3}
      >
        <Field
          component={TextField}
          fullWidth
          label={<FormattedMessage id="ui-organizations.summary.accountingCode" />}
          name="erpCode"
          validateFields={[]}
        />
      </Col>
      <Col
        xs={6}
        md={3}
      >
        <Field
          component={Select}
          fullWidth
          label={<FormattedMessage id="ui-organizations.summary.organizationStatus" />}
          name="status"
          placeholder=" "
          required
          validate={validateRequired}
          validateFields={[]}
        >
          {Object.keys(ORGANIZATION_STATUS).map((key) => (
            <FormattedMessage
              id={`ui-organizations.organizationStatus.${key}`}
              key={key}
            >
              {(message) => <option value={ORGANIZATION_STATUS[key]}>{message}</option>}
            </FormattedMessage>
          ))}
        </Field>
      </Col>
      <Col
        xs={6}
        md={3}
      >
        <FieldLanguage
          labelId="ui-organizations.summary.defaultLanguage"
          name="language"
        />
      </Col>
      <Col
        xs={6}
        md={3}
      >
        <Field
          component={Checkbox}
          label={<FormattedMessage id="ui-organizations.summary.isVendor" />}
          name="isVendor"
          type="checkbox"
          onChange={onChangeIsVendor}
          vertical
          validateFields={[]}
        />
      </Col>
      <Col
        xs={6}
        md={3}
      >
        <AcqUnitsField
          id="org-acq-units"
          name="acqUnitIds"
          perm={isEditMode ? MANAGE_UNITS_PERM : CREATE_UNITS_PERM}
          isEdit={isEditMode}
          preselectedUnits={initialValues.acqUnitIds}
          isFinal
        />
      </Col>
      <Col
        xs={6}
        md={3}
      >
        <Field
          label={<FormattedMessage id="ui-organizations.summary.description" />}
          name="description"
          component={TextArea}
          validateFields={[]}
        />
      </Col>
      <Col xs={6}>
        <FieldArray
          addLabel={<FormattedMessage id="ui-organizations.summary.add" />}
          component={RepeatableField}
          id="aliases"
          legend={<FormattedMessage id="ui-organizations.summary.alternativeNames" />}
          name="aliases"
          renderField={renderAlias}
        />
      </Col>

      {isVendorUncheckConfirm && (
        <ConfirmationModal
          id="uncheck-is-vendor-confirmation"
          confirmLabel={<FormattedMessage id="ui-organizations.vendor.confirmation.confirm" />}
          heading={<FormattedMessage id="ui-organizations.vendor.confirmation.heading" />}
          message={<FormattedMessage id="ui-organizations.vendor.confirmation.message" />}
          onCancel={hideVendorUncheckConfirm}
          onConfirm={handleVendorUncheck}
          open
        />
      )}
    </Row>
  );
}

OrganizationSummaryForm.propTypes = {
  initialValues: PropTypes.object.isRequired,
};

export default OrganizationSummaryForm;
