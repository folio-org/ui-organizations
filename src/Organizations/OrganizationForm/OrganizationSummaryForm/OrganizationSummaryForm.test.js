import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import stripesFinalForm from '@folio/stripes/final-form';

import { organization, organizationTypes } from '../../../../test/jest/fixtures';
import OrganizationSummaryForm from './OrganizationSummaryForm';

import { useTypes } from '../../../common/hooks';

jest.mock('../../../common/hooks', () => ({
  useTypes: jest.fn(),
}));

const TestForm = stripesFinalForm({})(
  () => {
    return (
      <form>
        <OrganizationSummaryForm
          initialValues={organization}
        />
      </form>
    );
  },
);

const renderForm = ({ initialValues = {} } = {}) => render(
  <TestForm
    onSubmit={jest.fn()}
    initialValues={initialValues}
  />,
  { wrapper: MemoryRouter },
);

describe('OrganizationSummaryForm', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
    useTypes
      .mockClear()
      .mockReturnValue({ organizationTypes, totalRecords: organizationTypes.length });
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure', async () => {
    const { asFragment } = renderForm({ initialValues: organization });

    await screen.findByText('ui-organizations.summary.name');

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render active type', async () => {
    renderForm({ initialValues: organization });

    expect(document.querySelector('#multiselect-option-list-organizations-type')).toBeInTheDocument();
    expect(document.querySelectorAll('#multiselect-option-list-organizations-type [role=option]').length).toEqual(1);
  });

  it('should show the label of the selected initial active value type once', async () => {
    renderForm({ initialValues: organization });

    const selectedType = await screen.findAllByText('Book trade');

    expect((selectedType).length).toEqual(1);
  });
});
