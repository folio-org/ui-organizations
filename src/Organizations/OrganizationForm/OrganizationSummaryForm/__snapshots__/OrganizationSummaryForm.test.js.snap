// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationSummaryForm should render correct structure 1`] = `
<DocumentFragment>
  <form>
    <div
      class="row"
    >
      <div
        class="col-xs-6 col-md-3"
      >
        <div
          class="textField"
        >
          <label
            class="label"
            for="name"
          >
            ui-organizations.summary.name
            <span
              aria-hidden="true"
              class="asterisk"
            >
              *
            </span>
          </label>
          <div
            class="formControl inputGroup"
          >
            <input
              aria-invalid="false"
              aria-required="true"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              error=""
              id="name"
              name="name"
              required=""
              spellcheck="false"
              type="text"
              value="Amazon.com"
              warning=""
            />
          </div>
          <div
            role="alert"
          />
        </div>
      </div>
      <div
        class="col-xs-6 col-md-3"
      >
        <div
          class="textField"
        >
          <label
            class="label"
            for="text-input-1"
          >
            ui-organizations.summary.code
            <span
              aria-hidden="true"
              class="asterisk"
            >
              *
            </span>
          </label>
          <div
            class="formControl inputGroup"
          >
            <input
              aria-invalid="false"
              aria-required="true"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              error=""
              id="text-input-1"
              name="code"
              required=""
              spellcheck="false"
              type="text"
              value="AMAZ"
              warning=""
            />
          </div>
          <div
            role="alert"
          />
        </div>
      </div>
      <div
        class="col-xs-6 col-md-3"
      >
        <div
          class="textField"
        >
          <label
            class="label"
            for="text-input-2"
          >
            ui-organizations.summary.accountingCode
          </label>
          <div
            class="formControl inputGroup"
          >
            <input
              aria-invalid="false"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              error=""
              id="text-input-2"
              name="erpCode"
              spellcheck="false"
              type="text"
              value="G64758-74834"
              warning=""
            />
          </div>
          <div
            role="alert"
          />
        </div>
      </div>
      <div
        class="col-xs-6 col-md-3"
      >
        <div
          class="select inputGroup fullWidth"
        >
          <label
            class="label"
            for="select-3"
            id="select-3-label"
          >
            ui-organizations.summary.organizationStatus
            <span
              aria-hidden="true"
              class="asterisk"
            >
              *
            </span>
          </label>
          <div
            class="selectWrap"
          >
            <select
              aria-invalid="false"
              aria-labelledby="select-3-label"
              aria-required="true"
              class="selectControl formControl"
              id="select-3"
              name="status"
              required=""
            >
              <option
                disabled=""
                value=""
              >
                 
              </option>
              <option
                value="Active"
              >
                ui-organizations.organizationStatus.active
              </option>
              <option
                value="Inactive"
              >
                ui-organizations.organizationStatus.inactive
              </option>
              <option
                value="Pending"
              >
                ui-organizations.organizationStatus.pending
              </option>
            </select>
            <span>
              Icon
            </span>
          </div>
          <div
            role="alert"
          />
        </div>
      </div>
      <div
        class="col-xs-6 col-md-3"
      >
        <div
          class="tether-target tether-enabled tether-abutted tether-abutted-left tether-abutted-right tether-abutted-top tether-abutted-bottom tether-element-attached-top tether-element-attached-center tether-target-attached-bottom tether-target-attached-center"
          style="position: relative; width: 100%;"
        >
          <div
            aria-live="assertive"
            aria-relevant="additions"
            class="sr-only"
          />
          <label
            class="label"
            for="stripes-selection-4"
            id="sl-label-stripes-selection-4"
          >
            ui-organizations.summary.defaultLanguage
          </label>
          <div
            class="selectionControlContainer"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="sl-label-stripes-selection-4 selected-stripes-selection-4-item"
              aria-owns="sl-undefined"
              class="selectionControl formControl"
              id="stripes-selection-4"
              name="language"
              type="button"
            >
              <span
                class="sr-only"
              >
                stripes-components.selection.controlLabel
              </span>
              <div
                class="singleValue"
                id="selected-stripes-selection-4-item"
              />
            </button>
            <div
              class="selectionEndControls"
            >
              <div
                class="textFieldIcon"
              >
                <span>
                  Icon
                </span>
              </div>
            </div>
          </div>
          <div
            role="alert"
          />
        </div>
      </div>
      <div
        class="col-xs-6 col-md-3"
      >
        <div
          class="checkbox inline vertical"
          data-test-checkbox="true"
        >
          <label
            class="label label checkboxInteractionStylesControl"
            for="checkbox-5"
          >
            <span
              class="labelText"
            >
              ui-organizations.summary.isVendor
            </span>
            <span
              class="inner checkboxInteractionStyles"
            >
              <input
                aria-invalid="false"
                checked=""
                class="input"
                id="checkbox-5"
                name="isVendor"
                type="checkbox"
                value=""
              />
              <span
                class="checkboxIcon"
              >
                <svg
                  height="10"
                  viewBox="0 0 14 14"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.395 4.617L5.63 9.383 3.605 7.358"
                    fill="none"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                    stroke-width="1.5"
                  />
                </svg>
              </span>
            </span>
          </label>
          <div
            role="alert"
          />
        </div>
      </div>
      <div
        class="col-xs-6 col-md-3"
      >
        <div
          class="multiSelectContainer"
          id="org-acq-units"
        >
          <div
            aria-live="assertive"
            aria-relevant="additions"
            class="sr-only"
          />
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="org-acq-units-label"
            class="multiSelectContainer"
            role="application"
          >
            <label
              class="label"
              for="org-acq-units-input"
              id="org-acq-units-label"
            >
              stripes-acq-components.label.acqUnits
            </label>
            <span
              class="sr-only"
              id="multi-value-status-org-acq-units"
            >
              0 items selected
            </span>
            <span
              class="sr-only"
              id="multi-describe-control-org-acq-units"
            >
              stripes-components.multiSelection.controlDescription
            </span>
            <div
              class="multiSelectControlWrapper"
            >
              <div
                aria-describedby="multi-describe-control-org-acq-units"
                aria-labelledby="org-acq-units-label multi-value-status-org-acq-units"
                class="multiSelectControl formControl"
                tabindex="0"
              >
                <div
                  class="multiSelectControlGroup"
                >
                  <input
                    aria-autocomplete="list"
                    aria-labelledby="org-acq-units-label"
                    autocomplete="off"
                    class="multiSelectFilterField"
                    id="org-acq-units-input"
                    type="text"
                    value=""
                  />
                  <input
                    aria-autocomplete="list"
                    aria-hidden="true"
                    autocomplete="off"
                    class="multiSelectValueInput"
                    id="multiselect-input-org-acq-units"
                    tabindex="-1"
                    type="text"
                    value=""
                  />
                </div>
                <button
                  aria-haspopup="true"
                  aria-label="open menu"
                  class="multiSelectToggleButton"
                  data-toggle="true"
                  role="button"
                  type="button"
                >
                  <div
                    class="textFieldIcon"
                  >
                    <span>
                      Icon
                    </span>
                  </div>
                </button>
              </div>
              <div
                class="multiSelectMenu"
                hidden=""
                style="width: 100px;"
              >
                <div
                  class="multiSelectFeedback"
                  role="alert"
                >
                  <div
                    class="multiSelectEmptyMessage"
                  >
                    stripes-components.multiSelection.defaultEmptyMessage
                  </div>
                </div>
                <ul
                  aria-labelledby="org-acq-units-label"
                  class="multiSelectOptionList"
                  id="multiselect-option-list-org-acq-units"
                  role="listbox"
                  style="max-height: 168px;"
                />
              </div>
            </div>
          </div>
          <div
            role="alert"
          />
        </div>
      </div>
      <div
        class="col-xs-6 col-md-3"
      >
        <div
          class="textArea inputGroup"
        >
          <label
            class="label"
            for="textarea-input-6"
          >
            ui-organizations.summary.description
          </label>
          <textarea
            aria-invalid="false"
            class="formControl"
            id="textarea-input-6"
            name="description"
            type="text"
          >
            Use for rush orders, replacement requests, and second hand market
          </textarea>
          <div
            role="alert"
          />
        </div>
      </div>
      <div
        class="col-xs-6"
      >
        <fieldset
          class="repeatableField hasMargin"
          data-test-repeatable-field="true"
          id="aliases"
        >
          <legend
            class="headline size-medium margin-medium font-weight-bold"
            data-test-headline="true"
            data-test-repeatable-field-legend="true"
          >
            ui-organizations.summary.alternativeNames
          </legend>
          <ul
            class="repeatableFieldList"
            data-test-repeatable-field-list="true"
          >
            <li
              class="repeatableFieldItem"
              data-test-repeatable-field-list-item="true"
            >
              <div
                class="repeatableFieldItemContent"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-xs"
                    data-test-alias="true"
                  >
                    <div
                      class="textField"
                    >
                      <label
                        class="label"
                        for="text-input-7"
                      >
                        ui-organizations.summary.alias
                        <span
                          aria-hidden="true"
                          class="asterisk"
                        >
                          *
                        </span>
                      </label>
                      <div
                        class="formControl inputGroup"
                      >
                        <input
                          aria-invalid="false"
                          aria-required="true"
                          autocapitalize="off"
                          autocomplete="off"
                          autocorrect="off"
                          error=""
                          id="text-input-7"
                          name="aliases[0].value"
                          required=""
                          spellcheck="false"
                          type="text"
                          value="Amazon"
                          warning=""
                        />
                      </div>
                      <div
                        role="alert"
                      />
                    </div>
                  </div>
                  <div
                    class="col-xs"
                  >
                    <div
                      class="textField"
                    >
                      <label
                        class="label"
                        for="text-input-8"
                      >
                        ui-organizations.summary.description
                      </label>
                      <div
                        class="formControl inputGroup"
                      >
                        <input
                          aria-invalid="false"
                          autocapitalize="off"
                          autocomplete="off"
                          autocorrect="off"
                          error=""
                          id="text-input-8"
                          name="aliases[0].description"
                          spellcheck="false"
                          type="text"
                          value=""
                          warning=""
                        />
                      </div>
                      <div
                        role="alert"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="repeatableFieldRemoveItem "
              >
                <button
                  aria-label="stripes-components.deleteThisItem"
                  class="iconButton medium"
                  data-test-repeatable-field-remove-item-button="true"
                  icon="trash"
                  type="button"
                >
                  <span
                    class="iconButtonInner mediumOffset"
                    data-test-repeatable-field-remove-item-button="true"
                  >
                    <span>
                      Icon
                    </span>
                  </span>
                </button>
              </div>
            </li>
          </ul>
          <button
            class="button default"
            data-test-repeatable-field-add-item-button="true"
            id="aliases-add-button"
            type="button"
          >
            <span
              class="inner"
            >
              ui-organizations.summary.add
            </span>
          </button>
        </fieldset>
      </div>
    </div>
  </form>
  <div />
</DocumentFragment>
`;
